### PR TITLE
Second day

### PR DESCRIPTION
**Pull request by hrodberaht**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,64 +1,77 @@
...
+    });
+  }
 
   onShowAddCard = () => {
     const { showAddCard } = this.state;
     this.setState({ showAddCard: !showAddCard });
   };
 
-  onAddCard = (card) => {
+  onAddCard = card => {
```
Comment (ar-mac) [inline src/App.js:41]: Can you describe what's the difference between setting this method as fat arrow and regular function `onAddCard(card) {` ?

File: src/App.js [hunk 2]
```diff
@@ -1,64 +1,77 @@
...
     const { cards, lastId } = this.state;
-    cards.push({...card,id: lastId+1}); //TODO find another way to generate id
-    console.log(card);
-
-    this.setState({cards: cards, lastId: lastId+1});
-  }
+    cards.unshift({ ...card, id: lastId + 1 });
+    this.setState({ cards: cards, lastId: lastId + 1 });
```
Comment (ar-mac) [inline src/App.js:44]: We can use shorthand notation if our key, and variable assigned to it are the same
`this.setState({ cards, lastId: lastId + 1 });`
Also we can generate unique id on multiple different ways `Date.now()` is the simplest one.

File: src/App.js [hunk 3]
```diff
@@ -1,64 +1,77 @@
...
 
-  onAddCard = (card) => {
+  onAddCard = card => {
     const { cards, lastId } = this.state;
-    cards.push({...card,id: lastId+1}); //TODO find another way to generate id
-    console.log(card);
-
-    this.setState({cards: cards, lastId: lastId+1});
-  }
+    cards.unshift({ ...card, id: lastId + 1 });
```
Comment (ar-mac) [inline src/App.js:43]: Unshift mutates the initial array. We should not mutate `state` and `props` for some reasons I can explain if you wish.
The easiest way of updating the array is to use destructurization.
Something like
`const newCards = [{...card, id: lastid + 1}, ...cards]`

File: src/App.js [hunk 4]
```diff
@@ -1,64 +1,77 @@
...
+    this.setState({ cards: cards, lastId: lastId + 1 });
+  };
 
-  onDeleteCard = (id) => {
+  onDeleteCard = id => {
     const { cards } = this.state;
-    const index = cards.findIndex((card) => {
+    const index = cards.findIndex(card => {
```
Comment (ar-mac) [inline src/App.js:49]: With fat arrow we can use shoarthand notation in situation when it immediately returns result  `card => card.id === id`

File: src/components/AllCards/AllCards.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    }
+
+    return (
+      <div>
+        {list.map(card => {
+          return (
+            <Card
+              key={card.id}
+              card={card}
+              handleClick={id => handleClick(id)}
```
Comment (ar-mac) [inline src/components/AllCards/AllCards.js:21]: In such cases we can pass the function directly `handleClick={handleClick}` no neeed to wrap it into fat arrow. It also has minor performance cost, as inline fat arrows are recreated on every render.

File: src/App.js (hunk 2 above)
Comment (hrodberaht) [reply src/App.js:44]: ok

File: src/App.js (hunk 3 above)
Comment (hrodberaht) [reply src/App.js:43]: ok

File: src/App.js (hunk 1 above)
Comment (hrodberaht) [reply src/App.js:41]: It's about this, if we don't use fat arrow, we must bind function with this like this.onShowAddCard.bind.(this)

**Pull request merged**